### ISSUE TITLE
Push command and button push to wrong branch?

### ISSUE THREAD

**Issue by borisghidaglia** [gitkraken/vscode-gitlens#3227]
### Description

Hello 👋

> [!important] 
> This seems to be too big of a problem to be happening for everyone. Moreover, I tested to rollback to previous versions but the problem persisted. Thus, I suppose I'm currently in a weird edge case.

I thought I was going crazy after pushing to dev "by mistake", but I juste realized the button in the picture isn't related to my current branch. I suppose the same problem goes for the command (from command palette) but I'm not sure how to verify it.

<img width="703" alt="image" src="https://github.com/gitkraken/vscode-gitlens/assets/26144896/e7c98fba-0339-4a63-afa5-0b54560346da">

Until a few days ago (I believe), the behavior wasn't this one. It was pushing to my current branch, and displaying push or pull on this button based on my current branch.

### GitLens Version

14.9.0

### VS Code Version

Version: 1.88.0 (Universal)
Commit: 5c3e652f63e798a5ac2f31ffd0d863669328dc4c
Date: 2024-04-03T13:28:18.899Z (1 wk ago)
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Darwin arm64 23.4.0

### Git Version

git version 2.39.3 (Apple Git-146)

**Comment by jkelroy** [comment]
Hello @borisghidaglia, can you expand on exactly what is not working as expected? Is the issue that it shows push to _origin/dev_ (when that is not the actual upstream), that pushing will push to another branch besides _origin/dev_ or that origin/dev is being set as the upstream unexpectedly?

Notably, if you create a branch in GitLens, the upstream will be automatically set to the upstream of the branch you created off of - we have a feature request to prompt for this [here](https://github.com/gitkraken/vscode-gitlens/issues/2778). So if the issue is that the upstream is being set to main expectedly for you, then the linked feature request should help solve for this.

**Comment by borisghidaglia** [comment]
Hello @jkelroy, thanks for your answer :) 

> Is the issue that it shows push to origin/dev (when that is not the actual upstream) [...]

Yes exactly. It pushed to `origin/dev` when I expected it to push to `origin/feature/my-feature`, as my current branch was `feature/my-feature`.

From the issue you're sharing: 

> [...] when creating a branch from a remote only branch [...]

Thanks I understand now, it is probably what I did. I tried and indeed the upstream is set to the upstream of the branch I created off of.

But why is this the default behavior?
